odl/solvers/nonsmooth/adupdates.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Alternating dual (AD) update algorithm studied by McGaffin and Fessler.
+
+The alternating dual upgrade method is reported to be a flexible method which
+allows an acceleration by permitting efficient calculations on a GPU.
+"""
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+__all__ = ('adupdates', 'adupdates_simple')
+
+
+def adupdates(x, g, L, stepsize, majs, niter, random=False,
+              callback=None, callback_loop='outer'):
+    """Alternating Dual updates method.
+
+    The Alternating Dual (AD) updates method of McGaffin and Fessler `[MF2015]
+    <http://ieeexplore.ieee.org/document/7271047/>`_ is designed to solve an
+    optimization problem of the form
+
+        min_x sum_i g_i(L_i x)
+
+    where ``g_i`` are proper, convex and lower semicontinuous functions and
+    ``L_i`` are linear `Operator`s.
+
+    Parameters
+    ----------
+    g : sequence of `Functional`s
+        All functions need to provide a `convex_conj` with a
+        `proximal` factory.
+    L : sequence of `Operator`s
+        Needs to provide as many elements as `g`.
```

Comment:
This sentence could use some clarification. What is provided? Typically we say that the length of the sequence should be something